CHANGELOG.md

```diff
@@ -37,6 +37,7 @@ For older changes see the [archived Singularity change log](https://github.com/a
 - Adding a new remote endpoint using the `apptainer remote add` command will
   now set the new endpoint as default. This behavior can be suppressed by
   supplying the `--no-default` (or `-n`) flag to `remote add`.
+- Template supports skipping comment lines started with a hash and a space `#␣`.
```

Comment:
This implies that it only covers lines that begin with hash-space, but actually it also applies to comments later on in the line, right?  I think the CHANGELOG entry should be more accurate.

Also, does it really have anything to do with templating?  Doesn't it apply to all definition files, whether or not templating is being used?

Now I'm wondering under what conditions exactly that comments aren't already being applied. At least in sections that are interpreted by a shell (e.g. %post) the shell should be removing them.  Maybe this shouldn't be being applied to the entire definition file, because a shell would also be smart enough to ignore comments that are inside strings, for example.

Reply:
yeah, good point, any places having comment line pattern will be erased. 
I also think that this comment line erase feature should only not be applied to all sections.

Reply:
good points, thanks for the comments.
I've updated the entry in the CHANGELOG. I also refactored the code. The current plan is to check whether the template `{{ VAR }}` is defined inside a comment line during template parsing. If it is defined inside a comment line `# `, then the replacement will be skipped.